.github/workflows/libc-overlay-tests.yml

```diff
@@ -0,0 +1,92 @@
+# This workflow is for pre-commit testing of the LLVM-libc project.
+name: LLVM-libc Pre-commit Overlay Tests
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'libc/**'
+      - '.github/workflows/libc-overlay-tests.yml'
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      # Set fail-fast to false to ensure that feedback is delivered for all matrix combinations.
+      fail-fast: false
+      matrix:
+        include:
+          # TODO: add linux gcc when it is fixed
+          - os: ubuntu-24.04
+            compiler:
+              c_compiler: clang
+              cpp_compiler: clang++
+          - os: windows-2022
+            compiler:
+              c_compiler: clang-cl
+              cpp_compiler: clang-cl
+          - os: macos-14
+            compiler:
+              c_compiler: clang
+              cpp_compiler: clang++
+    
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup ccache
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        max-size: 1G
+        key: libc_overlay_build_${{ matrix.os }}_${{ matrix.compiler.c_compiler }}
+        variant: sccache
+    
+    - name: Prepare dependencies (Ubuntu)
+      if: runner.os == 'Linux'
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libmpfr-dev libgmp-dev libmpc-dev ninja-build
+    
+    - name: Prepare dependencies (Windows)
+      if: runner.os == 'Windows'
+      run: |
+        choco install ninja
```

Comment:
What is `choco`?

https://github.com/chocolatey/choco

does that itself have to be installed, or does it ship on the github buildbots? I guess same question for `brew` on OSX; usually those don't come pre-installed IIRC.

Reply:
It is shipped in image per github spec. Please see https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md